### ISSUE TITLE
Would be nice if total counts can be displayed in a small box before or after the buttons

### ISSUE THREAD

**Issue by bboerendans** [kni-labs/rrssb#49]
Total sharing counts can be tracked with www.sharedcount.com. So we can show something like:

![naamloos](https://cloud.githubusercontent.com/assets/4182238/3972496/023d0d04-27de-11e4-96ad-ca7f5824724b.png)

I have no idea how to fix this, but i thought maybe its a nice idea  :)

**Comment by rjgamer** [comment]
+1, great idea!

**Comment by dbox** [comment]
![share-counter image](https://cloud.githubusercontent.com/assets/4182238/3972496/023d0d04-27de-11e4-96ad-ca7f5824724b.png)
I'm okay with adding this as an option, if anyone is up for taking it on. Will need to be added to 2.0 branch.

**Comment by connorwyatt** [comment]
I'll have a look into this, I have a good idea of how to get it done. I did something similar before I found RRSSB.

**Comment by dbox** [comment]
Awesome. If you do get something going, would you mind pulling into 2.0 branch?

**Comment by dbox** [comment]
Want to keep this separate for a bit until we can fix #75 as part of the last big 1.x

**Comment by connorwyatt** [comment]
No worries I will do and I'll have a look at #75 too

**Comment by connorwyatt** [comment]
@dbox I'm looking at the js and I think it might be best to do a rewrite of it. The functions would stay in it but it would end up as a plugin where you would call it like this (with more options of course):

```
<body>

  <div class="rrssb-container></div>

  <script>

    var options = {
      socialNetworks: [ "facebook", "twitter", "linkedin", "email" ],
      showCount: true
    };

    $(".rrssb-container").rrssb();

  </script>

</body>
```

What do you think of that?

The social network buttons would be created by the js plugin by specifying them in the order you want them to appear in the options.socialNetworks property.

Might allow some more customisability?

**Comment by dbox** [comment]
I think this makes a lot of sense. Would definitely be easier for someone to implement. The only downside is not being able to manually edit the meta info in html, but I think that is probably okay. 

I'm going to start a new 2.0 thread to capture everything.

**Comment by connorwyatt** [comment]
I'm glad, I started working on it a little bit today and I will implement
optional overrides so you can feed in your own twitter status or Facebook
status etc. Otherwise it will use the page's title and description along
with the url

**Comment by dbox** [comment]
The Meta already has settings in the .js...

**Comment by connorwyatt** [comment]
What do you mean?

**Comment by dbox** [comment]
```
// Settings that $.rrssb() will accept.
  var settings = $.extend({
    description: undefined,
    emailAddress: undefined,
    emailBody: undefined,
    emailSubject: undefined,
    image: undefined,
    title: undefined,
    url: undefined
  }, options );
```

**Comment by dbox** [comment]
I guess that can get called with the plugin.

**Comment by connorwyatt** [comment]
Yeah thats where I'm going to put them all. I'll add in other options for changing the text shown in individual social networks' buttons.

So for example there will also be an option for linkedinTitle that will override the default which will be the page title, along with others

**Comment by connorwyatt** [comment]
So if you dont want the linked in title to be the page name you can call it like this:

```javascript
$(".rrssb-container").rrssb({
  linkedinTitle: "This is a different title for LinkedIn only."
});
```

**Comment by connorwyatt** [comment]
But all others (FB, twitter etc.) will use the default. This way you can change the share message

**Comment by dbox** [comment]
It should be 2-tiered, though right?

If nothing set > use current page meta. If title set> Used `title` for `linkedinTitle`, then if `linkedinTitle` set, obviously use that.

That being said, I'm not even sure if I care about specific titles per sites. Not sure anyone would even need that.

**Comment by connorwyatt** [comment]
Yeah that's exactly right, it is 2-tiered, and that's not a problem, by default it will use the page title and description from the html

**Comment by dbox** [comment]
While I'm thinking about this: I think it's okay to have the share-count defaulted to visible, but only if it has 1 or more share(s).

**Comment by connorwyatt** [comment]
It’s a good idea, but then you have the overhead of http requests that are never used. I can do it if you want though

**Comment by dbox** [comment]
Np. Let's table for now

**Comment by mmautner** [comment]
I'm testing out the 2.0 branch with a very simple configuration:

```
    var options = {
       socialNetworks: ["linkedin", "facebook", "twitter", "reddit"],
       url: "some-valid-url",
       showCount: true,
    };
    $(".rrssb-container").rrssb(options);
```

And am noticing the Twitter API that's hit [no longer appears to be supported circa 2015](https://twittercommunity.com/t/how-to-get-proper-twitter-share-count-for-a-url/53876/2):

<img width="1214" alt="screen shot 2017-12-13 at 8 29 06 pm" src="https://user-images.githubusercontent.com/1419652/33975622-761d14ea-e044-11e7-9e2a-000eee3393ef.png">

Is this library still aiming to support share counts? Or maybe should it be dropped from 2.0?

Thank you @dbox for the work you've put into it so far 👍
